tree-sitter-stack-graphs/src/lib.rs

```diff
@@ -0,0 +1,550 @@
+//! # // graph content).  This minimizes the amount of work that we do when running `cargo test`.
+//! # mod tree_sitter_python {
+//! #   pub use ::tree_sitter_python::language;
+//! #   pub static STACK_GRAPH_RULES: &str = "";
+//! # }
+//! #
+//! # fn main() -> Result<(), Box<dyn std::error::Error>> {
+//! let python_source = r#"
+//!   import sys
+//!   print(sys.path)
+//! "#;
+//! let grammar = tree_sitter_python::language();
+//! let tsg_source = tree_sitter_python::STACK_GRAPH_RULES;
+//! let mut language = StackGraphLanguage::new(grammar, tsg_source)?;
+//! let mut stack_graph = StackGraph::new();
+//! let file_handle = stack_graph.get_or_create_file("test.py");
+//! language.load_stack_graph(&mut stack_graph, file_handle, python_source)?;
+//! # Ok(())
+//! # }
+//! ```
+
+use controlled_option::ControlledOption;
+use lsp_positions::SpanCalculator;
+use stack_graphs::arena::Handle;
+use stack_graphs::graph::File;
+use stack_graphs::graph::Node;
+use stack_graphs::graph::NodeID;
+use stack_graphs::graph::StackGraph;
+use thiserror::Error;
+use tree_sitter::Parser;
+use tree_sitter_graph::functions::Functions;
+use tree_sitter_graph::graph::Graph;
+use tree_sitter_graph::graph::GraphNode;
+use tree_sitter_graph::graph::GraphNodeRef;
+use tree_sitter_graph::graph::Value;
+use tree_sitter_graph::Identifier;
+use tree_sitter_graph::Variables;
+
+/// Holds information about how to construct stack graphs for a particular language
+pub struct StackGraphLanguage {
+    parser: Parser,
+    tsg: tree_sitter_graph::ast::File,
+    functions: Functions,
+}
+
+impl StackGraphLanguage {
+    /// Creates a new stack graph language, loading in the TSG graph construction rules from a
+    /// string.
+    pub fn new(
+        language: tree_sitter::Language,
+        tsg_source: &str,
+    ) -> Result<StackGraphLanguage, LanguageError> {
+        let mut parser = Parser::new();
+        parser.set_language(language)?;
+        let mut tsg = tree_sitter_graph::ast::File::new(language);
+        tsg.parse(tsg_source)?;
+        let functions = Functions::stdlib();
+        Ok(StackGraphLanguage {
+            parser,
+            tsg,
+            functions,
+        })
+    }
+}
+
+/// An error that can occur while loading in the TSG stack graph construction rules for a language
+#[derive(Debug, Error)]
+pub enum LanguageError {
+    #[error(transparent)]
+    LanguageError(#[from] tree_sitter::LanguageError),
+    #[error(transparent)]
+    ParseError(#[from] tree_sitter_graph::ParseError),
+}
+
+impl StackGraphLanguage {
+    /// Executes the graph construction rules for this language against a source file, creating new
+    /// nodes and edges in `stack_graph`.  Any new nodes that we create will belong to `file`.
+    /// (The source file must be implemented in this language, otherwise you'll probably get a
+    /// parse error.)
+    pub fn load_stack_graph(
```

Comment:
That's a good point — `StackGraphLoader` is using "load" correctly, since we will have already executed the graph DSL at that point.  Maybe "build" instead of "load" here?